web/docs/auth/migrate-from-11.md

```diff
@@ -0,0 +1,318 @@
+1. You can now delete the migration script and the `db.seeds` config
+
+
+## Migrating a Deployed App
+
+To migrate a deployed app, you need to go through the migration steps while deploying the app in between the steps.
+
+1. Go through the steps of [Migrate to the New Auth System](#1-migrate-to-the-new-auth-system) on your local machine _until the step where you run the migration script_
+2. Deploy the app (#1)
+3. Run the migration script on the production database
+
+    :::info Running the migration script on the production database
+
+    If you have an app that is already deployed, you need to run the migration script on the production database.
+
+    We wrote instructions on how to do it for **Fly.io** deployments here: https://github.com/wasp-lang/wasp/issues/1464 
+
+    The instructions should be similar for other deployment providers: setting up a tunnel from your local machine to the production database and running the migration script locally with `DATABASE_URL` set to the production database.
+    :::
+4. Finish the steps of [Migrate to the New Auth System](#1-migrate-to-the-new-auth-system) on your local machine
+5. Deploy the app (#2)
+6. Go through the steps of [Cleanup the Old Auth System](#2-cleanup-the-old-auth-system) on your local machine
+7. Deploy the app (#3)
+
+Your app should now be migrated to the new auth system.
+
+## Example Migration Scripts
+
+### Username & Password
+
+```ts title="src/server/migrateToNewAuth.ts"
+import { PrismaClient } from "@prisma/client";
+import { ProviderName, UsernameProviderData } from "@wasp/auth/utils";
+
+export async function migrateUsernameAuth(prismaClient: PrismaClient) {
+  const users = await prismaClient.user.findMany({
+    include: {
+      auth: true,
+    },
+  });
+
+  for (const user of users) {
+    if (user.auth) {
+      console.log("User was already migrated, skipping", user);
+      continue;
+    }
+
+    const providerData: UsernameProviderData = {
+      hashedPassword: user.password,
+    };
+    const providerName: ProviderName = "username";
+
+    await prismaClient.auth.create({
+      data: {
+        identities: {
+          create: {
+            providerName,
+            providerUserId: user.username.toLowerCase(),
+            providerData: JSON.stringify(providerData),
+          },
+        },
+        user: {
+          connect: {
+            id: user.id,
+          },
+        },
+      },
+    });
+  }
+}
+```
+
+
+### Email
+
+```ts title="src/server/migrateToNewAuth.ts"
+import { PrismaClient } from "@prisma/client";
+import { EmailProviderData, ProviderName } from "@wasp/auth/utils";
+
+export async function migrateEmailAuth(prismaClient: PrismaClient) {
+  const users = await prismaClient.user.findMany({
+    include: {
+      auth: true,
+    },
+  });
+
+  for (const user of users) {
+    if (user.auth) {
+      console.log("User was already migrated, skipping", user);
+      continue;
+    }
+
+    if (!user.email || !user.password) {
```

Comment:
You don't have this same check for username in the script above, why not?

Reply:
Good catch 👍